scanner/src/test/java/edu/ucr/cs/riple/scanner/tools/SerializationTestHelper.java

```diff
@@ -86,17 +99,28 @@ public SerializationTestHelper<T> setOutputFileNameAndHeader(String fileName, St
     return this;
   }
 
-  public void doTest() {
+  public void prepareTest() {
     Preconditions.checkNotNull(factory, "Factory cannot be null");
     Preconditions.checkNotNull(fileName, "File name cannot be null");
-    Path outputPath = outputDir.resolve(fileName);
+    outputFilePath = outputDir.resolve(fileName);
     try {
-      Files.deleteIfExists(outputPath);
+      Files.deleteIfExists(outputFilePath);
     } catch (IOException ignored) {
-      throw new RuntimeException("Failed to delete older file at: " + outputPath);
+      throw new RuntimeException("Failed to delete older file at: " + outputFilePath);
     }
+  }
+
+  public void doTest(Class<? extends Exception> exception, String expectedErrorMessage) {
```

Comment:
👍 

Appreciate that we separated the two versions of this method. Looking at it now, can I suggest renaming this one `doTestExpectingError(...)`, just to make it even more obvious what is happening when reading the test cases.